Update navigation bar to match cilium.io

Kikiodazie opened the pull request:
What this PR does is to update the current https://docs.cilium.io/ sphinx theme navigation bar to match the cilium.io's in solving this issue https://github.com/cilium/cilium/issues/19171

I am not much of a frontend/UI developer, so I approached the issue to the best of my knowledge and kept things simple by changing only the `sphinx_rtd_theme/layout.html` file. 

My code isn't the best as it introduces some dependencies ([w3.css](https://www.w3schools.com/w3css/defaulT.asp), Iconify and fontawesome) but maybe this could work in the interim or be iterated on and the issue left open till a more experienced frontend developer makes the code and design better.

Kikiodazie commented:
Hey @qmonnet, don't know if you've had the time to check on this issue.

I hope this meets you well. Cheers.

qmonnet commented:
Right sorry, this slipped through, apologies. I'll definitely look at it this week. Thanks for the reminder!

Kikiodazie commented:
> Right sorry, this slipped through, apologies. I'll definitely look at it this week. Thanks for the reminder!

No problem @qmonnet. Thank you. Do have a great week ahead.

qmonnet reviewed (changes requested):
Looks good overall, with a few observations. My main questions are:

- Do we really need to include 3 more resources? Maybe some of them are not really necessary?
- Question for @lizrice: How close to cilium.io did you want this header to be? Did you mean a few adjustments so they look closer, or did you want ~100% identical headers?

Kikiodazie commented on sphinx_rtd_theme/layout.html:
```diff
@@ -180,13 +193,30 @@
         </li>
-        <li><a href="https://cilium.io">Cilium</a></li>
-        <li><a href="https://cilium.io/blog">Blog</a></li>
         <li><a href="https://docs.cilium.io">Documentation</a></li>
```
True. I just followed what's on the cilium.io, but I don't think we should.

Kikiodazie commented on sphinx_rtd_theme/layout.html:
```diff
@@ -180,13 +193,30 @@
-            <span>Enterprise</span>
-          </a>
+          <div style="background-color: white; font-size: 16px; font-weight: 600; margin-left: 40px" class="w3-dropdown-hover">
+            <button class="w3-button">Learn<i class="fa fa-angle-down"></i></button>
```
Ok then. Makes sense. Noted.

Kikiodazie replied:
Added spacing with the new pointing down triangle. 

What do you think? does the one in the screenshot work?

![Screenshot 2022-06-20 at 11 37 52](https://user-images.githubusercontent.com/42554056/174596000-9770a2dd-4994-47cc-9aed-881a84c66522.png)

Kikiodazie commented on sphinx_rtd_theme/layout.html:
```diff
@@ -180,13 +193,30 @@
-            <span>Enterprise</span>
-          </a>
+          <div style="background-color: white; font-size: 16px; font-weight: 600; margin-left: 40px" class="w3-dropdown-hover">
+            <button class="w3-button">Learn<i class="fa fa-angle-down"></i></button>
```
Yess unicode. good idea. Thanks will take your feedback that into account.

Kikiodazie replied:
I have removed the font-awesome library and used the pointing down triangle on ebpf.io instead. And it looks much better.
![Screenshot 2022-06-20 at 11 37 52](https://user-images.githubusercontent.com/42554056/174584253-a2b372eb-8103-4d29-8912-877ac75b4c78.png)

Kikiodazie commented on sphinx_rtd_theme/layout.html:
```diff
@@ -172,6 +175,16 @@
+                target="_blank"
+              >
+                <span class="iconify" data-icon="logos:slack-icon"></span>
+                <span class="gh-text">Join Slack</span>
```
Will work on that

Kikiodazie replied:
Removed Iconify 
![Screenshot 2022-06-20 at 12 38 21](https://user-images.githubusercontent.com/42554056/174595524-9b419692-066a-4232-aacf-bcb73e61a26b.png)
and Added horizontal space (margin).

Kikiodazie commented on sphinx_rtd_theme/layout.html:
```diff
@@ -63,6 +65,7 @@
   {%- endif -%}
 
   {#- JAVASCRIPTS #}
+  <script src="https://code.iconify.design/2/2.2.1/iconify.min.js"></script>
```
for the slack icon.

Kikiodazie commented:
> Looks good overall, with a few observations. My main questions are:
> 
> * Do we really need to include 3 more resources? Maybe some of them are not really necessary?
> * Question for @lizrice: How close to cilium.io did you want this header to be? Did you mean a few adjustments so they look closer, or did you want ~100% identical headers?

Thanks for the feedback @qmonnet  will work on them and update you as soon as possible.

Kikiodazie commented on sphinx_rtd_theme/layout.html:
```diff
@@ -172,6 +175,16 @@
+                rel="noopener noreferrer"
+                target="_blank"
+              >
+                <span class="iconify" data-icon="logos:slack-icon"></span>
```
Thank you @lizrice

Kikiodazie replied:
Done. I have removed iconify and used the inline svg @lizrice shared. 

See the new look:
![Screenshot 2022-06-20 at 12 38 21](https://user-images.githubusercontent.com/42554056/174595699-94b7e7db-be41-46c7-b1eb-27f66e77e52b.png)

Kikiodazie commented on sphinx_rtd_theme/layout.html:
```diff
@@ -180,13 +193,30 @@
+            <div style="margin-left: -15px" class="w3-dropdown-content w3-bar-block w3-card-4">
+              <a href="https://cilium.io/get-started" class="w3-bar-item w3-button">Get Started</a>
+              <a href="https://cilium.io/get-involved" class="w3-bar-item w3-button">Get Involved</a>
+              <a href="https://cilium.io/get-help" class="w3-bar-item w3-button">Get Help</a>
```
I have implemented this feedback.

Here is a link to the commit: https://github.com/cilium/sphinx_rtd_theme/pull/18/commits/37be2b6164921e7bb005e026ae9a806f01631084

Here's how it looks now, What do you think.
![Screenshot 2022-06-20 at 16 55 48](https://user-images.githubusercontent.com/42554056/174640053-2e2b7ccf-faba-4aae-872e-15e3dc53a76f.png)
![Screenshot 2022-06-20 at 16 56 04](https://user-images.githubusercontent.com/42554056/174640096-88453d74-131a-4769-95aa-284cf616ff31.png)

Kikiodazie commented:
Hey @qmonnet @lizrice,

Thank you very much for your valuable feedback, I have implemented all of them. 

Here is how the NavBar looks now:

<img width="1440" alt="Screenshot 2022-06-20 at 16 57 50" src="https://user-images.githubusercontent.com/42554056/174640721-e977cef2-6c69-4dc3-af1d-3de440c6310a.png">

What do you think? 

I cut it down to only 1 resource. Sorry my HTML, CSS and frontend skills  are not the best. I hope this works. 

Let me know if there's any other thing I need to do.

qmonnet reviewed (changes requested):
Looks great, nice work!

Please find some additional suggestions inline below.

Also:
- Can you please update the link for “Documentation”? It doesn't point to the `stable` branch like in the cilium.io header.
- Would be great if we could also have the text of the different items turn blue when the user hover with the mouse, just like on cilium.io. This is not a blocker though.

Kikiodazie commented:
> Looks great, nice work!
> 
> Please find some additional suggestions inline below.
> 
> Also:
> 
> * Can you please update the link for “Documentation”? It doesn't point to the `stable` branch like in the cilium.io header.
> * Would be great if we could also have the text of the different items turn blue when the user hover with the mouse, just like on cilium.io. This is not a blocker though.

Thanks for the suggestion @qmonnet. Will work on them and update you when I do.

Kikiodazie commented on sphinx_rtd_theme/layout.html:
```diff
@@ -162,16 +163,28 @@
+                target="_blank"
+                style="margin-left: 15px;"
+              >
+                <img style="height: 1em;" src="https://raw.githubusercontent.com/cilium/cilium.io/main/src/icons/slack.inline.svg" alt="slack_icon">
```
Hey @qmonnet, I added the SVGs locally, but there rose a new issue. 

I couldn’t get it to render even when I am accessing the right file.
<img width="575" alt="Screenshot 2022-07-03 at 17 27 59" src="https://user-images.githubusercontent.com/42554056/177048644-b6cfebb4-42be-45b9-8dee-8d0988c2b149.png">

I spent hours trying to fix it using different HTML tags as it seems to be a known issue with SVGs but to no avail.

I then thought of a way around it:

Since the goal is for it not to depend on the cilium.io repository, I put the src for each svg to the link it will be hosted on the theme repo. Similar to this -> `https://raw.githubusercontent.com/Kikiodazie/cilium_sphinx_rtd_theme/pr/add-svg-locally/docs/_static/icons/slack.inline.svg`

I tested it with my fork and it works just fine. What do you think about it? Though it works, if you can help me figure out getting the SVG to render without the hassle that would be perfect. 

I hope this meets you well.

qmonnet replied:
Of the two solutions, I would much rather have inline SVGs. Using the full URL to the GH repo is so likely to break and be hard to debug if we ever rename a branch or a repo at some point.

But if you didn't manage to link the files, this is maybe simply because they were not copied from the theme repo to the `_static` directory for the docs along with the CSS. Have you tried adding your `icons` directory to [`package_data`](https://github.com/Kikiodazie/cilium_sphinx_rtd_theme/blob/v1.0/setup.py#L117)?

Kikiodazie replied:
So i tried adding the `icons` directory to the [package_data](https://github.com/Kikiodazie/cilium_sphinx_rtd_theme/blob/v1.0/setup.py#L117), and it didn't work. 

So I just added the inline SVGs. Do check it out and let me know if it works.

qmonnet replied:
Thanks, will have a look next week :+1:

Kikiodazie replied:
No problem. Thank you. Do have a great weekend.

qmonnet replied:
So adding the SVGs locally works just fine for me. Here's what I had to do:

- I reused your PR but discarded the last commit (so I'm in the state where you added the icons locally but it didn't work, and before you added them inline in the HTML).
- I moved the SVGs from `docs/_static/icons` to `sphinx_rtd_theme/static/icons`. This is where they should be. I probably got you confused when mentioning `Documentation/_static` in my comment below, but that's where I wanted to see them end up when building the docs in the Cilium repo, not in _this_ repo. Apologies for not being more explicit.
- I added `static/icons/*.svg` to `package_data` in `setup.py`
- I fixed the paths in `sphinx_rtd_theme/layout.html`, respecting the templating syntax, to make it point to the correct location.

This is available on [this branch](https://github.com/qmonnet/sphinx_rtd_theme/commits/test_svgs), see the last commit.

Kikiodazie replied:
Thank you very much for this walk-through, Quentin, I sincerely appreciate it. 

Thank you for the branch I just went ahead and rebased it on this my branch 😁🫣

Kikiodazie commented on sphinx_rtd_theme/layout.html:
```diff
@@ -180,13 +193,51 @@
+            <button class="w3-button">Learn <span class="triangle">▾</span></button>
+            <div class="w3-dropdown-content w3-bar-block w3-card-4">
+              <a style = "margin-left: 0px;" href="https://cilium.io/get-started" class="w3-bar-item w3-button">
+                <img style="height: 1.5em;" src="https://raw.githubusercontent.com/cilium/cilium.io/main/src/icons/get-started.inline.svg" alt="get-started-icon">
```
Done 

Check commit https://github.com/cilium/sphinx_rtd_theme/pull/18/commits/31320a830828e85633ff48e7e451e7d208688119

Kikiodazie commented:
> Looks great, nice work!
> 
> Please find some additional suggestions inline below.
> 
> Also:
> 
> * Can you please update the link for “Documentation”? It doesn't point to the `stable` branch like in the cilium.io header.
> * Would be great if we could also have the text of the different items turn blue when the user hover with the mouse, just like on cilium.io. This is not a blocker though.

Hey @qmonnet, I have made your suggested changes and left a comment regarding an issue adding SVG locally.

![Screenshot 2022-07-03 at 19 06 25](https://user-images.githubusercontent.com/42554056/177052277-90b40ba4-5bf4-40ef-b2ee-0c1685e80af9.png)
![Screenshot 2022-07-03 at 19 06 14](https://user-images.githubusercontent.com/42554056/177052282-978a07f8-3bbb-4bcc-90e3-fca5cf9ec952.png)

qmonnet commented on sphinx_rtd_theme/layout.html:
```diff
@@ -180,13 +193,51 @@
+            <button class="w3-button">Learn <span class="triangle">▾</span></button>
+            <div class="w3-dropdown-content w3-bar-block w3-card-4">
+              <a style = "margin-left: 0px;" href="https://cilium.io/get-started" class="w3-bar-item w3-button">
+                <img style="height: 1.5em;" src="https://raw.githubusercontent.com/cilium/cilium.io/main/src/icons/get-started.inline.svg" alt="get-started-icon">
```
`Documentation/_static` referred to where they'd land when building the docs within the Cilium repository, sorry for the confusion.

Kikiodazie replied:
Thank you very much for the clarification. ❤️

qmonnet reviewed (changes requested):
Sorry for the delay, I finally took some time to look at it again this morning.

Let's remove the inline SVGs and include them locally, it's not hard to do, see [my comment above](https://github.com/cilium/sphinx_rtd_theme/pull/18#discussion_r951343494).

Testing locally (with the SVG inclusion fix, but I don't think it should matter), I see that the “Get involved” link doesn't fit on the row, can we make the menu a tad wider?

![image](https://user-images.githubusercontent.com/17001771/185914616-460daff7-cbc4-4c49-9da7-a10295b99edf.png)

We'll also need to squash all these commits.

Kikiodazie commented:
Hey @qmonnet,

So so sorry for getting to this late 🙏🏾.

I have implemented all your requested changes and squashed all the commits into one. 

I hope this meets you well. Do have a great rest of the day. 

Regards,
Divine Odazie.

xmulligan commented:
One final request, can you update the Enterprise link to point to https://cilium.io/enterprise/ rather than isovalent.com?

Kikiodazie commented:
> One final request, can you update the Enterprise link to point to https://cilium.io/enterprise/ rather than isovalent.com?

Hey @xmulligan, 

Thanks for this request, but the enterprise link already points to https://cilium.io/enterprise/ as in the screenshot below.
<img width="635" alt="Screenshot 2022-09-21 at 17 00 28" src="https://user-images.githubusercontent.com/42554056/191554030-e753025d-a4ba-4c6f-9baa-8c73f5bb9df3.png">

xmulligan commented:
ahh perfect thanks! Seems that was pulled through from Cilium.io and I missed it 💯

qmonnet reviewed (approved):
OK good I tested again and it looks perfect this time, and this has been pending for way too long. Thanks Divine for bearing with me on this one :) Let's get this merged!

Can you please handle the follow-up PR in Cilium? We need to update the commit hash in Documentation/requirements.txt for the change to take effect on the builds.

Kikiodazie commented:
> OK good I tested again and it looks perfect this time, and this has been pending for way too long. Thanks Divine for bearing with me on this one :) Let's get this merged!
> 
> Can you please handle the follow-up PR in Cilium? We need to update the commit hash in Documentation/requirements.txt for the change to take effect on the builds.

Will do that right away. Thank you very much Quentin for your guidance. I sincerely appreciate it.

Kikiodazie commented:
Done https://github.com/cilium/cilium/pull/21436.

Thank you once again Quentin.
